src/ansys/sherlock/core/launcher.py

```diff
@@ -132,13 +143,26 @@ def connect_grpc_channel(port=SHERLOCK_DEFAULT_PORT):
     return SHERLOCK
 
 
-def _get_base_ansys():
+def _get_base_ansys(year=None, release_number=None):
     supported_installed_versions = {
         env_key: path
         for env_key, path in os.environ.items()
         if env_key.startswith("AWP_ROOT") and os.path.isdir(path)
     }
 
+    try:
+        if year and release_number:
+            year = _extract_sherlock_version_year(year)
+            version_key = f"AWP_ROOT{year}{release_number}"
+            if version_key in supported_installed_versions:
+                return supported_installed_versions[version_key]

```

Comment:
I think if the year and version are specified but not installed, this should cause an error indicating that the specified version is not installed. Otherwise, the user may not realize that their intended version of Sherlock isn't being launched. Consider the case where they typed the wrong value.